### PR TITLE
[pallet-revive] eth-rpc fixes

### PR DESCRIPTION
**Pull request by pgherveou**
- Breaking down the integration-test into multiple tests
- Fix tx hash to use expected keccak-256
- Add option to ethers.js example to connect to westend and use a private key

### CODE REVIEW

**Comment by pgherveou** [comment]
cc @xermicus @athei

File: prdoc/pr_6453.prdoc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+title: '[pallet-revive] breakdown integration tests'
+doc:
+- audience: Runtime Dev
+  description: null
```
Comment (pgherveou) [inline prdoc/pr_6453.prdoc:4]:
```suggestion
  description: Break down the single integration tests into multiple tests
```

File: prdoc/pr_6453.prdoc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+title: '[pallet-revive] breakdown integration tests'
+doc:
+- audience: Runtime Dev
+  description: Break down the single integration tests into multiple tests
```
Comment (pgherveou) [inline prdoc/pr_6453.prdoc:4]:
```suggestion
  description: Break down the single integration tests into multiple tests, use keccak-256 for tx.hash
```

**Review by athei — APPROVED** [review]
I assume we will encounter similar problems with the block hash?

**Comment by smiasojed** [comment]
Can I assume that it is transparent for REMIX? it calls `eth_getTransactionReceipt` to check if transaction has been processed.

**Comment by pgherveou** [comment]
> `eth_getTransactionReceipt`

Yeah then it might be fine, unless it's doing similar checks to verify that the receipt tx hash match the tx hash from the payload it just sent

**Comment by athei** [comment]
So the problem arises when a library hashes the tx itself instead of just relying on the hash returned from `eth_sendRawTransaction`?

**Comment by pgherveou** [comment]
> So the problem arises when a library hashes the tx itself instead of just relying on the hash returned from `eth_sendRawTransaction`?

Yes it assert that the returned value matches the expected one

**Comment by pgherveou** [comment]
@smiasojed can you ✅ if that's good for you, or let me know if you have other questions

**Pull request merged**